### PR TITLE
133 update improvements

### PR DESCRIPTION
**Pull request by khoroshevskyi**

### CODE REVIEW

File: pepdbagent/db_utils.py [hunk 1]
```diff
@@ -133,10 +127,22 @@ class Samples(Base):
 
     id: Mapped[int] = mapped_column(primary_key=True)
     sample: Mapped[dict] = mapped_column(JSON, server_default=FetchedValue())
-    row_number: Mapped[int]
+    row_number: Mapped[int]  # TODO: should be removed
```
Comment (nleroy917) [inline pepdbagent/db_utils.py:130]: Do we need to remove this now?

File: pepdbagent/modules/project.py [hunk 2]
```diff
@@ -57,7 +51,8 @@ def get(
         namespace: str,
         name: str,
         tag: str = DEFAULT_TAG,
-        raw: bool = False,
+        raw: bool = True,
```
Comment (nleroy917) [inline pepdbagent/modules/project.py:54]: We default to `raw` now? What was the reason for the change?

File: pepdbagent/modules/project.py [hunk 3]
```diff
@@ -66,6 +61,7 @@ def get(
         :param name: name of the project (Default: name is taken from the project object)
         :param tag: tag (or version) of the project.
         :param raw: retrieve unprocessed (raw) PEP dict.
+        :param with_id: retrieve project with id [default: False]
```
Comment (nleroy917) [inline pepdbagent/modules/project.py:64]: Should we be more specific about what the `id` means? Like, maybe `guid` or `pephub id`? Or something?

File: pepdbagent/modules/project.py [hunk 4]
```diff
@@ -126,14 +118,42 @@ def get(
...
+        :param session: open session object
+        :param prj_id: project id
+        :param with_id: retrieve sample with id
+        """
+        samples_results = session.scalars(select(Samples).where(Samples.project_id == prj_id))
+        result_dict = {}
+        for sample in samples_results:
+            sample_dict = sample.sample
+            if with_id:
+                sample_dict["ph_id"] = sample.guid
```
Comment (nleroy917) [inline pepdbagent/modules/project.py:135]: Should this be a constant? And will this be confusing in the future?

Comment (nleroy917) [reply pepdbagent/modules/project.py:135]: Like, should we just always call it a `guid`?

File: pepdbagent/utils.py [hunk 5]
```diff
@@ -104,3 +105,53 @@ def convert_date_string_to_date(date_string: str) -> datetime.datetime:
...
+    # Find the Root Node
+    # Create a lookup dictionary for nodes by their GUIDs
+    guid_lookup = {entry["guid"]: entry for entry in results.values()}
+
+    # Create a dictionary to map each GUID to its child GUID
+    parent_to_child = {
+        entry["parent_guid"]: entry["guid"]
+        for entry in results.values()
+        if entry["parent_guid"] is not None
+    }
```
Comment (nleroy917) [inline pepdbagent/utils.py:131]: Could this ever result in multiple? Since its a `for-loop` and creating a list?

File: pepdbagent/utils.py [hunk 6]
```diff
@@ -104,3 +105,53 @@ def convert_date_string_to_date(date_string: str) -> datetime.datetime:
     :return: datetime format
     """
     return datetime.datetime.strptime(date_string, "%Y/%m/%d") + datetime.timedelta(days=1)
+
+
+def order_samples(results: dict) -> List[dict]:
```
Comment (nleroy917) [inline pepdbagent/utils.py:110]: I trust this works, is this something worth optimizing in a lower-level if it needs to really be fast?

**Review by nleroy917 — APPROVED** [review]
Some comments, but I think it all looks clean and well-put together

File: pepdbagent/modules/project.py (hunk 2 above)
Comment (khoroshevskyi) [reply pepdbagent/modules/project.py:54]: We almost never use processed pep, so I decided to change it to raw

File: pepdbagent/modules/project.py (hunk 4 above)
Comment (khoroshevskyi) [reply pepdbagent/modules/project.py:135]: Yeah, I missed it

File: pepdbagent/utils.py (hunk 6 above)
Comment (khoroshevskyi) [reply pepdbagent/utils.py:110]: yes, if we want it to be super fast, we should write it in Rust

File: pepdbagent/db_utils.py (hunk 1 above)
Comment (khoroshevskyi) [reply pepdbagent/db_utils.py:130]: yes, but in the future, after database will be done

File: pepdbagent/utils.py (hunk 5 above)
Comment (khoroshevskyi) [reply pepdbagent/utils.py:131]: ??
it's creating dict from list results, it's impossible that we will have 2 parent_guid same

**Pull request merged**